httpcore5-h2/src/main/java/org/apache/hc/core5/http2/priority/PriorityParamsParser.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * ====================================================================
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * ====================================================================
+ *
+ * This software consists of voluntary contributions made by many
+ * individuals on behalf of the Apache Software Foundation.  For more
+ * information on the Apache Software Foundation, please see
+ * <http://www.apache.org/>.
+ *
+ */
+package org.apache.hc.core5.http2.priority;
+
+import java.util.Locale;
+
+import org.apache.hc.core5.annotation.Internal;
+import org.apache.hc.core5.util.CharArrayBuffer;
+import org.apache.hc.core5.util.Tokenizer;
+
+@Internal
+public final class PriorityParamsParser {
+    private static final Tokenizer TK = Tokenizer.INSTANCE;
+    private static final Tokenizer.Delimiter KEY_DELIMS = Tokenizer.delimiters('=', ',', ';');
+    private static final Tokenizer.Delimiter VALUE_DELIMS = Tokenizer.delimiters(',', ';');
+
+    private PriorityParamsParser() {
+    }
+
+    public static PriorityParams parse(final String headerValue) {
+        Integer urgency = null;     // null means “u absent”
+        Boolean incremental = null; // null means “i absent”
+
+        if (headerValue == null || headerValue.isEmpty()) {
+            return new PriorityParams(null, null);
+        }
+
+        final CharArrayBuffer buf = new CharArrayBuffer(headerValue.length());
```

Comment:
@arturobernalg Please do not do that. There is no need to create a copy of the header value. The parser can work with any `CharSequence`.  Can't you re-use `MessageSupport` parsing routines here?

Otherwise, looks good.

Reply:
@ok2c  please do another pass